client/src/components/Upload/DefaultBox.vue

```diff
@@ -99,7 +100,16 @@ const listExtensions = computed(() => props.effectiveExtensions.filter((ext) =>
 const showHelper = computed(() => Object.keys(uploadItems.value).length === 0);
 const uploadValues = computed(() => Object.values(uploadItems.value));
 
-const queue = computed(() => createUploadQueue());
+onMounted(() => {
+    queue.value = initUploadQueue();
+});
+
+watch(
+    () => props.historyId,
+    () => {
+        queue.value = initUploadQueue();
```

Comment:
Is this not a problem for uploads targeting another history ?

Reply:
I would have assumed you need to associate each upload with a history

Reply:
We are still associating the whole queue with a history. The only difference is that before we always created a new queue whenever the current history changed, but now we will re-target to the new history only IF the queue was not "started". If it was already running, it will continue running in the old history and we create a "new queue" for the "new history" so whatever we add next is going to be associated with the new one.

So, in summary, once we press "Start" whatever is in the queue will go to the current history if I didn't misunderstand anything.

Reply:
But you lost the old queue then, how does the progress bar update ?

Reply:
In my local tests the progress "works". I uploaded 100 files to "history A" and then switched to "history B" in the middle of the upload. The progress keeps going and the files get uploaded to the history A. The only thing is that, if you open the uploader, you need to wait until the progress is completed before you can try to upload anything to "history B". The UI won't let you otherwise.

If this is not the desired behavior I can try to change the implementation to maintain a unique queue and associate the history with each individual item when the user presses "Start" as you suggest.

Reply:
I will keep exploring a bit the option to associate the history with individual items and see if I can make it work without too many changes, if not, we can go with these changes as a fix :+1:

Reply:
Actually, this is a cleaner and simpler solution! Thanks for suggesting it!

It now lets you pause and schedule upload datasets to the new history and continue. Better user experience I think.

![UploadSwithHistoryFix](https://github.com/user-attachments/assets/6f9c1bbd-49a0-4a63-9713-3da4953ec7d2)

Reply:
It might just be me but isn't it a little difficult to decipher the difference between what will go to `Old History` and what to `New History`? As in, would it nicer to have an `<hr>` between the file queues in the uploader?

Reply:
This is a good point and we should definitely consider how to communicate this better in the future uploader redesign or in dev :+1: 
That said, I would prefer to keep the fix as minimal as possible, I already stretched it a bit by adding the history name at the top :stuck_out_tongue_closed_eyes:

Reply:
Fair point, we can keep the change minimal for now